### PR TITLE
Using recursion calculates the power(x, n).

### PR DESCRIPTION
**Pull request by anupamahalder**

### CODE REVIEW

File: Leetcode/50/Power_function.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/**
+ * 50. Pow(x,n) 
+ * (https://leetcode.com/problems/powx-n/description/)
+*/
+
+#include<bits/stdc++.h>
```
Comment (nitishhsinghhh) [inline Leetcode/50/Power_function.cpp:6]: Relevant headers

Comment (anupamahalder) [reply Leetcode/50/Power_function.cpp]: Yes sorry for that

Comment (nitishhsinghhh) [reply Leetcode/50/Power_function.cpp]: Np..

**Review by nitishhsinghhh — APPROVED** [review]
LGTM. 

Can you further optimise this?
Consider only one variable, n, we can break down the problem into three cases: when n is less than 0, when n is equal to 0, and when n is greater than 0. 
For the case when n is greater than 0, we can use a simple recursive approach to multiply half of the result of raising x to the power of n/2 with itself for even n, and further multiply by x for odd n. The case when n is equal to 0 is straightforward. 

There is a catch with n less than 0. This exact question was actually asked in a real interview.

**Comment by nitishhsinghhh** [comment]
class Solution {
public:
    double myPow(double x, int n) {
        if (n < 0) {
            return -------    ;
        }

        if (n == 0) 
            return ------;

      double half = myPow(x, n / 2);
      double ans = half * half;
      if (n % 2 == 0)
           return half * half;
      else
           return half * half * x;
    }
};

**Pull request merged**